MSAL/src/native_auth/public/state_machine/error/RetrieveAccessTokenError.swift

```diff
@@ -24,38 +24,39 @@
 
 import Foundation
 
-@objc
+@objcMembers
 public class RetrieveAccessTokenError: MSALNativeAuthError {
-    /// An error type indicating the type of error that occurred
-    @objc public let type: RetrieveAccessTokenErrorType
+    let type: RetrieveAccessTokenErrorType
 
     init(type: RetrieveAccessTokenErrorType, message: String? = nil) {
         self.type = type
-        super.init(message: message)
+        super.init(identifier: type.rawValue, message: message)
     }
 
+    /// Describes an error that provides messages describing why an error occurred and provides more information about the error.
     public override var errorDescription: String? {
-        if let description = super.errorDescription {
-            return description
-        }
-
-        switch type {
-        case .generalError:
-            return "General error"
-        case .refreshTokenExpired:
-            return "Refresh token expired"
-        case .tokenNotFound:
-            return "Token not found"
-        case .browserRequired:
-            return "Browser required"
-        }
+        return super.errorDescription ?? type.rawValue
```

Comment:
I don't think we should be mixing errorDescription, which is intended to be a human-friendly brief description of the error, with the rawValue of the type, which should be a more computer-friendly, short, fixed string identifier for the type, e.g. `token_not_found` (or even possibly `tokenNotFound`).

Reply:
Agreed (as [here](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1923#discussion_r1401117647))